### PR TITLE
fix: #51842

### LINKED ISSUE #51842: salt.auth.django docs use wrong model name
`salt.auth.django` doc uses:

```python
class SaltExternalAuthModel(models.Model):
    user_fk = models.ForeignKey(auth.User)
    minion_matcher = models.CharField()
    minion_fn = models.CharField()
```
as an exemple of a django model to auth users.

There's few problems:

- the actual module tries to to fetch `minion_or_fn_matcher` user attribute.

- `CharField` should have mandatory `max_length` kwarg.

- user foreign key should be:
  ```python
  from django.contrib.auth.models import User
  ...
  user_fk = models.ForeignKey(User, on_delete=models.CASCADE)
  ```

Docs should be updated to:

```python
from django.contrib.auth.models import User
class SaltExternalAuthModel(models.Model):
    user_fk = models.ForeignKey(User, on_delete=models.CASCADE)
    minion_or_fn_matcher = models.CharField(max_length=255)
    minion_fn = models.CharField(max_length=255)
```

### PR DESCRIPTION
**Pull request by mattLLVW**
### What does this PR do?

Fix #51842 Wrong model name
### What issues does this PR fix or reference?
 
#51842

### CODE REVIEW

File: salt/auth/django.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 .. code-block:: python
 
     class SaltExternalAuthModel(models.Model):
-        user_fk = models.ForeignKey(auth.User)
-        minion_matcher = models.CharField()
-        minion_fn = models.CharField()
+        user_fk = models.ForeignKey(User, on_delete=models.CASCADE)
+        minion_or_fn_matcher = models.CharField(max_length=255)
+        minion_fn = models.CharField(max_length=255)
```
Comment (waynew) [inline salt/auth/django.py]: If there's an assumption of Postgres as the database, [there's no reason to add max_length](https://www.depesz.com/2010/03/02/charx-vs-varcharx-vs-varchar-vs-text/).

Comment (waynew) [reply salt/auth/django.py]: Ah... I see it's [required in Django's CharField](https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.CharField)

Comment (waynew) [reply salt/auth/django.py]: Apparently one needs to [make their own `CharField` to avoid Django's limit](https://stackoverflow.com/a/20116327)

**Review by waynew — APPROVED** [review]

**Review by dwoz — CHANGES_REQUESTED** [review]
@mattLLVW Is it possible to add a test for this change?

**Comment by mattLLVW** [comment]
It's a documentation change...?

**Review by dwoz — APPROVED** [review]
@mattLLVW My bad I took a glance and it looked like a code change. Approved.

**Pull request merged**